Rogue-Ape-Studios/Crusader
# ✨ Item prototype

## Content

Created an item prototype system, for working with player stats.

## Changes

### Added
`Assets/Prefabs/Items.meta` : Was created.
`Assets/Scripts/Items.meta` : Was created.
`Assets/Prefabs/Items/Item.prefab` : Was created, and acts as a generic Item Prefab.
`Assets/Prefabs/Items/ItemManager.prefab` : Was created, and manages all item spawns.
`Assets/Scripts/Items/BasePlayerConfig.cs` : Was created, creates ScriptableObjects.
`Assets/Scripts/Items/CurrentPlayer.asset` : Was created, demo, but can also be used in future as the actual player stats.
`Assets/Scripts/Items/Item.cs` : Was created, the base class for all items.
`Assets/Scripts/Items/ItemManager.cs` : Was created, spawns, manages and applies item upgrades.
`Assets/Scripts/Items/ItemRarity.cs` : Was created, and allows the setting of item rarity.
`Assets/Scripts/Items/PassiveItem.cs` : Was created, the class for all passive (common) items.
`Assets/Scripts/Items/PlayerConfig.asset` : Was created.
`Assets/Scripts/Items/PlayerProperty.cs` : Was created. Dictates which property will be changed.

# Testing
1. Place the Item Manger and an Item into the scene.
2. Once the objects are set, use the context menu to spawn an object.
3. Use the player to tag the object.
4. Check the Players stats.

Closes #27
